### PR TITLE
Adding AnonymousUtils definition for parse

### PR DESCRIPTION
**Pull request by gnunicorn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://parseplatform.org/Parse-SDK-JS/api/2.18.0/Parse.AnonymousUtils.html
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## parse ([<kbd>unpkg</kbd>](https://unpkg.com/browse/parse@latest/))
was missing the following properties:
1. idempotency
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0c2934f59001dac2f429c7cf83ccb19548510bec
</p>

File: types/parse/index.d.ts [hunk 1]
```diff
@@ -1112,6 +1113,16 @@ namespace Parse {
...
     }
 
+    /**
+     * Provides utility functions for working with Anonymously logged-in users.
+     */
+    namespace AnonymousUtils  {
+        function init(): void;
+        function isLinked(user: User): boolean;
+        function link(user: User, options?: SuccessFailureOptions): Promise<User>;
+        function logIn(options?: FullOptions): Promise<User>;
```
Comment (RaschidJFR) [inline types/parse/index.d.ts:1123]:
```suggestion
        function logIn(options?: ScopeOptions): Promise<User>;
```

File: types/parse/index.d.ts [hunk 2]
```diff
@@ -1112,6 +1113,16 @@ namespace Parse {
         function track(name: string, dimensions: any): Promise<any>;
     }
 
+    /**
+     * Provides utility functions for working with Anonymously logged-in users.
+     */
+    namespace AnonymousUtils  {
+        function init(): void;
+        function isLinked(user: User): boolean;
+        function link(user: User, options?: SuccessFailureOptions): Promise<User>;
```
Comment (RaschidJFR) [inline types/parse/index.d.ts:1122]:
```suggestion
        function link(user: User, options?: ScopeOptions): Promise<User>;
```

File: types/parse/index.d.ts [hunk 3]
```diff
@@ -1112,6 +1113,16 @@ namespace Parse {
         function track(name: string, dimensions: any): Promise<any>;
     }
 
+    /**
+     * Provides utility functions for working with Anonymously logged-in users.
+     */
+    namespace AnonymousUtils  {
+        function init(): void;
```
Comment (RaschidJFR) [inline types/parse/index.d.ts:1120]: I can't find any reference to the `init()` function in the [source code](https://github.com/parse-community/Parse-SDK-JS/blob/master/src/AnonymousUtils.js). Should we remove this?
```suggestion
```

File: types/parse/parse-tests.ts [hunk 4]
```diff
@@ -240,6 +240,10 @@ function return_a_generic_query(): Parse.Query<Game> {
     return new Parse.Query(Game);
 }
 
+function test_anonymous_utils(): Promise<Parse.User> {
+    return Parse.AnonymousUtils.logIn({});
```
Comment (RaschidJFR) [inline types/parse/parse-tests.ts:244]:
```suggestion
    // $ExpectType boolean
    Parse.AnonymousUtils.isLinked(new Parse.User());
    // $ExpectType Promise<User<Attributes>>
    Parse.AnonymousUtils.link(new Parse.User(), { useMasterKey: true, sessionToken: '' });
    // $ExpectType Promise<User<Attributes>>
    Parse.AnonymousUtils.logIn({},  { useMasterKey: true, sessionToken: '' });
```

**Review by RaschidJFR — CHANGES_REQUESTED** [review]
Thanks for your work. Would you mind also adding these same changes to the the files inside the folder `ts3.6/` so your contributions are available for newer TS versions?

**Comment by RaschidJFR** [comment]
Hey @gnunicorn I see you've been busy, thanks a lot. It's late now to tell you, but for the next time it will save you time to run the tests on your end before pushing. Like this: `$ npm test parse`.

Just waiting for the bot to finish and we're good to go.

**Comment by gnunicorn** [comment]
yeah, running tests locally failed for other reasons, had to fight on the CI, sorry for the noise!

**Comment by gnunicorn** [comment]
@RaschidJFR anything else I should be doing?

**Review by RaschidJFR — APPROVED** [review]
LGTM

**Comment by gnunicorn** [comment]
Ready to merge

**Pull request merged**